lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystem.java

```diff
@@ -346,11 +347,14 @@ public Optional<UriLocation> encryptedPreSignedUri(Location location, Duration t
                 .requestPayer(requestPayer)
                 .key(s3Location.key())
                 .bucket(s3Location.bucket())
-                .applyMutation(builder -> key.ifPresent(encryption -> {
-                    builder.sseCustomerKeyMD5(md5Checksum(encryption));
-                    builder.sseCustomerAlgorithm(encryption.algorithm());
-                    builder.sseCustomerKey(encoded(encryption));
-                }))
+                .applyMutation(builder -> {
+                    key.ifPresent(encryption -> {
+                        builder.sseCustomerKeyMD5(md5Checksum(encryption));
+                        builder.sseCustomerAlgorithm(encryption.algorithm());
+                        builder.sseCustomerKey(encoded(encryption));
+                    });
+                    addEncryptionSettings(builder, context.s3SseContext());
```

Comment:
these are mutually exclusive - validate that we are not setting encryption keys twice